google-cloud-firestore/src/test/java/com/google/cloud/firestore/ToStringTest.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.firestore;
+
+import com.google.cloud.Timestamp;
+import com.google.cloud.firestore.spi.v1.FirestoreRpc;
+import java.util.Collections;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mockito;
+import org.mockito.Spy;
+import org.mockito.runners.MockitoJUnitRunner;
+
+/** @author Eran Leshem */
+@RunWith(MockitoJUnitRunner.class)
+public class ToStringTest {
+
+  @Spy
+  private final FirestoreImpl firestoreMock =
+      new FirestoreImpl(
+          FirestoreOptions.newBuilder().setProjectId("test-project").build(),
+          Mockito.mock(FirestoreRpc.class));
+
+  private WriteBatch batch;
+  private DocumentReference documentReference;
+
+  @Before
+  public void before() {
+    batch = firestoreMock.batch();
+    documentReference = firestoreMock.document("coll/doc");
+  }
+
+  @Test
+  public void testDocumentSnapshot() {
+    Assert.assertEquals(
```

Comment:
Testing the _exact_ return value of toString() is brittle (i.e. the test will fail with inconsequential changes to the return value of `toString()`; see https://abseil.io/resources/swe-book/html/ch12.html). Instead, please modify these tests to just check for "key information" in the string.

Also, please use `assertThat()` from `Truth` to do this, something like this, which IMO is more readable (note: untested code):

```java
String toStringResult = getDocumentSnapshot().toString();
assertThat(toStringResult).startsWith("DocumentSnapshot{");
assertThat(toStringResult).endsWith("}");
assertThat(toStringResult).contains("documents/coll/doc");
assertThat(toStringResult).contains("readTime=XXX");
```

Here is one other example from our test suite that uses a similar check: https://github.com/googleapis/java-firestore/blob/be5008f3de969df37932f8d844c8206597a59c14/google-cloud-firestore/src/test/java/com/google/cloud/firestore/it/ITQueryCountTest.java#L278-L282